### PR TITLE
[23.1] use router instead of hard links for certain page urls

### PR DESCRIPTION
**Pull request by martenson**
prevents refreshing, improves UX

## How to test the changes?
(Select all options that apply)
- [ ] I've included appropriate [automated tests](https://docs.galaxyproject.org/en/latest/dev/writing_tests.html).
- [x] This is a refactoring of components with existing test coverage.
- [ ] Instructions for manual testing are as follows:
  1. [add testing steps and prerequisites here if you didn't write automated tests covering all your changes]

## License
- [x] I agree to license these and all my past contributions to the core galaxy codebase under the [MIT license](https://opensource.org/licenses/MIT).

### CODE REVIEW

File: client/src/components/Page/PageDropdown.vue [hunk 1]
```diff
@@ -57,19 +62,31 @@ function onDelete(page_id: string) {
         </b-link>
         <p v-if="props.page.description">{{ props.page.description }}</p>
         <div class="dropdown-menu" aria-labelledby="page-dropdown">
-            <a class="dropdown-item dropdown-item-view" :href="urlView">
+            <a class="dropdown-item dropdown-item-view" href="javascript:void(0)" @click="pushUrl(urlView)">
```
Comment (guerler) [inline client/src/components/Page/PageDropdown.vue]: Thank you for refining this. We could use `router-link` here like we do e.g. in the admin panel: `<router-link to="/admin/sanitize_allow">Manage Allowlist</router-link>`. Then the context menu on links would work and the `computed` functions and `pushUrl` helper would not be needed.

Comment (martenson) [reply client/src/components/Page/PageDropdown.vue]: ohoho! nice, I'll do that

File: client/src/components/Page/PageList.vue [hunk 2]
```diff
@@ -61,8 +60,9 @@
                 </span>
                 <a
                     v-if="!row.item.published && !row.item.shared && !row.item.importable"
-                    :href="shareLink(row.item)"
-                    class="share-this-page">
+                    href="javascript:void(0)"
```
Comment (ElectronicBlueberry) [inline client/src/components/Page/PageList.vue]: Please avoid void links, as these are semantically incorrect. Having a void href makes this element no longer behave like a link, as it breaks on non-click actions.

You could either use a button styled like a link here, or you can have the click event prevent default behavior and keep the correct link in the href. With latter non left click actions (e.g. middle click, or copy link) will still preform as expected for links.

Comment (martenson) [reply client/src/components/Page/PageList.vue]: thanks for the catch!

**Review by jmchilton — APPROVED** [review]

**Pull request merged**